standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/common/histogram/KllHistogramEstimator.java

```diff
@@ -30,11 +30,11 @@ public class KllHistogramEstimator {
   private final KllFloatsSketch kll;
 
   KllHistogramEstimator() {
-    this.kll = new KllFloatsSketch();
+    this.kll = KllFloatsSketch.newHeapInstance();
```

Comment:
In this commit: https://github.com/apache/datasketches-java/commit/f3d80294e1e34278d1cd7a46071929ae267d633c#diff-255436ca709c4ad22450438c6b07cb84771cbb9694ebaec9b15ab201f8f03673

KllFloatsSketch was changed to `abstract`. Use `KllFloatsSketch.newHeapInstance()` to get the instance.

Reply:
dataskatches-hive 2.0 can read the data written by 1.2?

Reply:
> dataskatches-hive 2.0 can read the data written by 1.2?

I think yes.

Reply:
It's crucial to make sure backward compatibility, especially because stats in metastore are meant to be interoperable between engines, so if we are breaking compatibility with other engines using an older datasketch version, we must highlight it clearly.

I suggest to just do a quick test locally, compute with 1.2, print the quantiles, write to a file, load it with the current version and display the same. It can be just a blank java project with a main class, you don't even need Hive to test that, as we rely on the serde of the library.